lib/agent.js

```diff
@@ -369,6 +419,19 @@ Agent.prototype._handleMessage = function(request, callback) {
         return this._fetchSnapshot(request.c, request.d, request.v, callback);
       case 'nt':
         return this._fetchSnapshotByTimestamp(request.c, request.d, request.ts, callback);
+      case 'p':
+        var presence = this._createPresence(request);
```

Comment:
Might be a good idea to put this Agent presence-handling code behind a flag, so it's explicitly opt-in.

We know that this implementation doesn't necessarily scale to hundreds or thousands of clients on a single doc, so it's nice for a server to not have to worry about presence unless it's opted in.